pgx/src/datum/array.rs

```diff
@@ -20,7 +20,7 @@ use std::marker::PhantomData;
 use std::{mem, ptr, slice};
 
 pub struct Array<'a, T: FromDatum> {
-    ptr: Option<NonNull<pg_sys::varlena>>,
+    ptr: NonNull<pg_sys::varlena>,
     raw: Option<RawArray>,
```

Comment:
I want to have both of these stop being Options, but due to some other functions and the Drop implementation, that becomes a much larger cleanup job (the Option makes it easier to Drop the Array, basically). So, I opened this PR first because it can go separately.